addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -0,0 +1,54 @@
+odoo.define('website.s_table_of_content', function (require) {
+'use strict';
+
+const publicWidget = require('web.public.widget');
+
+const TableOfContent = publicWidget.Widget.extend({
+    selector: 'section .s_table_of_content_navbar_sticky',
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start: function () {
+        const $menuLoading = $('.o_menu_loading');
+        if (!$menuLoading.length) {
+            this._initializeNavbarTopPosition();
+        } else {
+            $(document.body).one('menu_loaded', () => this._initializeNavbarTopPosition());
+        }
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        this.$target.find('.s_table_of_content_navbar').css('top', '');
+        this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+    /**
+     * Initialize the top position of the snippet navbar according to the height
+     * of the headers of the page.
+     * @private
+     */
+    _initializeNavbarTopPosition: function () {
+        // TODO: find a way to adjust the horizontal navbar position according
+        // the header height/position when scrolling. (same issue whith s_scrollTo in snippets.animation) Event ?
```

Comment:
I am not sure to understand the problem reading the comment

Reply:
The problem is with the fixed header. It can appear or dissapear wiith the scroll top value. So if we define the top position of the horizontal navbar with the height of the header, it can be an empty space when header is not there during the scroll